packages/matter.js/src/cluster/definitions/OperationalCredentialsCluster.ts

```diff
@@ -0,0 +1,780 @@
+
+    /**
+     * This field shall be present whenever StatusCode has a value of OK. If present, it shall contain the Fabric Index
+     * of the Fabric last added, removed or updated.
+     *
+     * @see {@link MatterCoreSpecificationV1_1} § 11.17.6.10.2
+     */
+    fabricIndex: TlvOptionalField(1, TlvUInt8.bound({ min: 1, max: 254 })),
+
+    /**
+     * This field MAY contain debugging textual information from the cluster implementation, which SHOULD NOT be
+     * presented to user interfaces in any way. Its purpose is to help developers in troubleshooting errors and the
+     * contents MAY go into logs or crash reports.
+     *
+     * @see {@link MatterCoreSpecificationV1_1} § 11.17.6.10.3
+     */
+    debugText: TlvOptionalField(2, TlvString.bound({ maxLength: 128 }))
+});
+
+/**
+ * Input to the OperationalCredentials updateNoc command
+ *
+ * @see {@link MatterCoreSpecificationV1_1} § 11.17.6.9
+ */
+export const TlvUpdateNocRequest = TlvObject({
+    nocValue: TlvField(0, TlvByteString.bound({ maxLength: 400 })),
+    icacValue: TlvOptionalField(1, TlvByteString.bound({ maxLength: 400 }))
+});
+
+/**
+ * Input to the OperationalCredentials updateFabricLabel command
+ *
+ * @see {@link MatterCoreSpecificationV1_1} § 11.17.6.11
+ */
+export const TlvUpdateFabricLabelRequest = TlvObject({ label: TlvField(0, TlvString.bound({ maxLength: 32 })) });
+
+/**
+ * Input to the OperationalCredentials removeFabric command
+ *
+ * @see {@link MatterCoreSpecificationV1_1} § 11.17.6.12
+ */
+export const TlvRemoveFabricRequest = TlvObject({ fabricIndex: TlvField(0, TlvUInt8.bound({ min: 1, max: 254 })) });
+
+/**
+ * Input to the OperationalCredentials addTrustedRootCertificate command
+ *
+ * @see {@link MatterCoreSpecificationV1_1} § 11.17.6.13
+ */
+export const TlvAddTrustedRootCertificateRequest = TlvObject({
+    rootCaCertificate: TlvField(0, TlvByteString.bound({ maxLength: 400 }))
+});
+
+/**
+ * Operational Credentials
+ *
+ * This cluster is used to add or remove Node Operational credentials on a Commissionee or Node, as well as manage the
+ * associated Fabrics.
+ *
+ * @see {@link MatterCoreSpecificationV1_1} § 11.17
+ */
+export const OperationalCredentialsCluster = Cluster({
+    id: 0x3e,
+    name: "OperationalCredentials",
+    revision: 1,
+    features: {},
+
+    attributes: {
+        /**
+         * This attribute contains all NOCs applicable to this Node, encoded as a read-only list of NOCStruct.
+         *
+         * Operational Certificates shall be added through the AddNOC command, and shall be removed through the
+         * RemoveFabric command.
+         *
+         * Upon Factory Data Reset, this attribute shall be set to a default value of an empty list.
+         *
+         * The number of entries in this list shall match the number of entries in the Fabrics attribute.
+         *
+         * @see {@link MatterCoreSpecificationV1_1} § 11.17.5.1
+         */
+        noCs: FabricScopedAttribute(
```

Comment:
Hmm explain more please.  Maybe we can specialize at a different point in the stack?

Reply:
Marco created a FabricManager class which is holding "all data of a fabric". this includes nocs and such data too. it is already on my list to fix that because it also has issues for subscriptions, but in the end for now we store the fabricManager data, so it makes no sense to also store this here because this is currently not populated at all (and when we have a value that is also restored potentially leading to confusion maybe ... I would add an override with a Todo to clear that up for now ... This is a "because of current structure it is special).

Reply:
I see.  Since this is an implementation detail, WDYT about tweaking the cluster definition in CommissioningServer?  NBD if you'd prefer to hack the definition but would be nice to leave the clusters "standard".  I could make a convenient way of altering a cluster for use with a specific ClusterServer.

Reply:
OK, I've added the override.  I think in the future if we want to tweak the interfaces for our implementation using runtime changes I think that'd be OK but we can leave as an override for now.